operator/src/main/java/oracle/kubernetes/operator/steps/ManagedServerUpIteratorStep.java

```diff
@@ -182,13 +182,21 @@ public NextAction apply(Packet packet) {
     private boolean hasServerAvailableToStart(Packet packet) {
       DomainPresenceInfo info = packet.getSpi(DomainPresenceInfo.class);
       String adminServerName = ((WlsDomainConfig) packet.get(DOMAIN_TOPOLOGY)).getAdminServerName();
-      return ((numStarted.get() <= info.getNumScheduledManagedServers(clusterName, adminServerName)
+      return ((getNumServersStarted() <= info.getNumScheduledManagedServers(clusterName, adminServerName)
               && (canStartConcurrently(info.getNumReadyManagedServers(clusterName, adminServerName)))));
     }
 
     private boolean canStartConcurrently(long numReady) {
-      return ((this.maxConcurrency > 0) && (numStarted.get() < (this.maxConcurrency + numReady)))
-          || (this.maxConcurrency == 0);
+      return (ignoreConcurrencyLimits() ||
+              (this.maxConcurrency > 0) && (getNumServersStarted() < (this.maxConcurrency + numReady)));
```

Comment:
You should no longer need `(this.maxConcurrency > 0)` here, as you cannot reach this condition when it is 0. 

Can you explain the logic in the rest of the expression? Why can you start a new server when the number of servers started is less than the concurrency limit added to the number ready?

Reply:
ok, I'll make the change. As per OWLS-83431, we want to start one clustered server at a time and wait for it be scheduled before starting a new server in same cluster. So number of started servers in a cluster should be less than or equal to number of scheduled servers in that cluster. Initially when there are no servers scheduled, it'll start one server and wait for it be scheduled before starting second server. The scheduling happens very fast (few milli-seconds). Please see https://github.com/oracle/weblogic-kubernetes-operator/pull/1855#issuecomment-671665520 for more details.